### PR TITLE
cm-emmc-flashing: Update to refer to the mass-storage-gadget

### PR DESCRIPTION
**Pull request by timg236**
On CM4 and above the mass storage gadget is much faster than the previous VPU firmware USB MSD gadget implemenation. Point readers towards this first.

### CODE REVIEW

File: documentation/asciidoc/computers/compute-module/cm-emmc-flashing.adoc [hunk 1]
```diff
@@ -103,7 +103,10 @@ To set up software on a Windows host device:
 
 . Connect the IO Board to power. Windows should discover the hardware and configure the required drivers.
 
-. Double-click on `RPiBoot.exe` to run it. After a few seconds, the Compute Module eMMC should appear as a USB mass storage device.
+. On CM4 or CM5 run the 'Raspberry Pi - Mass Storage Gadget - 64-bit' from the start menu. After a few seconds, the Compute Module eMMC or NVMe will appear as USB mass storage devices. This also provides a debug console as a serial port gadget.
```
Comment (nathan-contino) [inline documentation/asciidoc/computers/compute-module/cm-emmc-flashing.adoc:106]:
```suggestion
. On CM4 and later devices, select 'Raspberry Pi - Mass Storage Gadget - 64-bit' from the start menu. After a few seconds, the Compute Module eMMC or NVMe will appear as USB mass storage devices. This also provides a debug console as a serial port gadget.
```

File: documentation/asciidoc/computers/compute-module/cm-emmc-flashing.adoc [hunk 2]
```diff
@@ -103,7 +103,10 @@ To set up software on a Windows host device:
...
+
+. On CM3 or older select 'rpiboot' Double-click on `RPiBoot.exe` to run it. After a few seconds, the Compute Module eMMC should appear as a USB mass storage device.
```
Comment (nathan-contino) [inline documentation/asciidoc/computers/compute-module/cm-emmc-flashing.adoc:108]:
```suggestion
. On CM3 or older devices, select 'rpiboot'. Double-click on `RPiBoot.exe` to run it. After a few seconds, the Compute Module eMMC should appear as a USB mass storage device.
```

**Review by nathan-contino — APPROVED** [review]
LGTM!

File: documentation/asciidoc/computers/compute-module/cm-emmc-flashing.adoc [hunk 3]
```diff
@@ -103,7 +103,10 @@ To set up software on a Windows host device:
 
 . Connect the IO Board to power. Windows should discover the hardware and configure the required drivers.
 
-. Double-click on `RPiBoot.exe` to run it. After a few seconds, the Compute Module eMMC should appear as a USB mass storage device.
+. On CM4 and later devices, select 'Raspberry Pi - Mass Storage Gadget - 64-bit' from the start menu. After a few seconds, the Compute Module eMMC or NVMe will appear as USB mass storage devices. This also provides a debug console as a serial port gadget.
+
+. On CM3 or older devices, select 'rpiboot'. Double-click on `RPiBoot.exe` to run it. After a few seconds, the Compute Module eMMC should appear as a USB mass storage device.
```
Comment (nathan-contino) [inline documentation/asciidoc/computers/compute-module/cm-emmc-flashing.adoc:108]:
```suggestion
. On CM4 and later devices, select **Raspberry Pi - Mass Storage Gadget - 64-bit** from the start menu. After a few seconds, the Compute Module eMMC or NVMe will appear as USB mass storage devices. This also provides a debug console as a serial port gadget.

. On CM3 and older devices, select **rpiboot**. Double-click on `RPiBoot.exe` to run it. After a few seconds, the Compute Module eMMC should appear as a USB mass storage device.
```

**Comment by lurch** [comment]
This updates the "on Windows" part of the instructions. Should we make a similar recommendation for the macOS and Linux instructions too?

**Comment by timg236** [comment]
> This updates the "on Windows" part of the instructions. Should we make a similar recommendation for the macOS and Linux instructions too?

The command line is slightly different but yes, the mass-storage-gadget64 is the preferred method for 2711 and 2712

**Pull request merged**